src/aiortc/contrib/media.py

```diff
@@ -100,62 +109,103 @@ def player_worker(
+                ):
+                    frame.pts = None
+                    frame = audio_resampler.resample(frame)
+
+                # fix timestamps
+                frame.pts = audio_samples
+                frame.time_base = fractions.Fraction(1, audio_sample_rate)
+                audio_samples += frame.samples
+
+                audio_fifo.write(frame)
+                while True:
+                    frame = audio_fifo.read(audio_samples_per_frame)
+                    if frame:
+                        frame_time = frame.time
+                        asyncio.run_coroutine_threadsafe(
+                            audio_track._queue.put(frame), loop
+                        )
+                    else:
+                        break
+            elif isinstance(frame, VideoFrame) and video_track:
+                if frame.pts is None:  # pragma: no cover
+                    logger.warning(
+                        "MediaPlayer(%s) Skipping video frame with no pts",
+                        container.name,
                     )
-                else:
-                    break
-        elif isinstance(frame, VideoFrame) and video_track:
-            if frame.pts is None:  # pragma: no cover
-                logger.warning(
-                    "MediaPlayer(%s) Skipping video frame with no pts", container.name
-                )
-                continue
+                    continue
+
+                # video from a webcam doesn't start at pts 0, cancel out offset
+                if video_first_pts is None:
+                    video_first_pts = frame.pts
+                frame.pts -= video_first_pts
 
-            # video from a webcam doesn't start at pts 0, cancel out offset
-            if video_first_pts is None:
-                video_first_pts = frame.pts
-            frame.pts -= video_first_pts
+                frame_time = frame.time
+                asyncio.run_coroutine_threadsafe(video_track._queue.put(frame), loop)
+        else:
+            try:
+                packet = next(container.demux(*streams))
+            except (av.AVError, StopIteration) as exc:
+                if isinstance(exc, av.FFmpegError) and exc.errno == errno.EAGAIN:
+                    time.sleep(0.01)
+                    continue
+                if audio_track:
+                    asyncio.run_coroutine_threadsafe(audio_track._queue.put(None), loop)
+                if video_track:
+                    asyncio.run_coroutine_threadsafe(video_track._queue.put(None), loop)
+                break
 
-            frame_time = frame.time
-            asyncio.run_coroutine_threadsafe(video_track._queue.put(frame), loop)
+            # read up to 1 second ahead
+            if throttle_playback:
+                elapsed_time = time.time() - start_time
+                if frame_time and frame_time > elapsed_time + 1:
+                    time.sleep(0.1)
+
+            if isinstance(*streams, AudioStream):
```

Comment:
This doesn't work if you have both audio and video streams.
```suggestion
            if isinstance(packet.stream, AudioStream):
```

Reply:
Please, could explain test to reproduce and test in my environment?

Reply:
Take some h264 encoded video and pass it to webcam example with `--play-from` and  `--no-transcode`.

Reply:
So, with this suggestion worked for you?

Reply:
Yep.